engine/src/api/utils/helpers.py

```
@@ -137,3 +138,94 @@ def clear_conversation_history(reason: str) -> Dict[str, Any]:
+    env_var_name = f"{provider}_API_KEY"
+    try:
+        # Use default=None to return None if the variable is not set
+        api_key = config(env_var_name, default=None)
+        if api_key:
+            logger.debug(f"Found API key for provider '{provider}' via env var {env_var_name}")
+            return api_key
+        else:
+            logger.warning(f"Environment variable {env_var_name} is set but empty.")
+            return None
+    except UndefinedValueError:
+        logger.warning(f"API key environment variable {env_var_name} not found for provider '{provider}'.")
```

Ruff (0.11.9)

Line too long (107 > 100)

(E501)